### PR TITLE
refactor: remove local_id from SwapSuccess

### LINKED ISSUE #733: remove local_id from `SwapResult` in nomatching mode
Currently nomatching mode (here a `SwapResult`) still maintains a `local_id` which doesn't make sense to me. There should be only one (global) `order_id`.

```
order_id: "631145c0-f901-11e8-b518-5fd626b71b77"
local_id: "e6fe35e0-f911-11e8-b9ca-27ecace2dafc"
pair_id: "LTC/BTC"
quantity: 0.001
r_hash: "aad8fae44138922fc52b1fca949c7f75294fa5f8345418634d23ba4d961b5b9f"
amount_received: 100000
amount_sent: 110000
peer_pub_key: "03fd337659e99e628d0487e4f87acf93e353db06f754dccc402f2de1b857a319d0"
currency_received: "LTC"
currency_sent: "BTC"
```

### PR DESCRIPTION
**Pull request by ImmanuelSegol**
closes #733 
Totally removes local_id according to:

> decision from the call: simply remove local_id from SwapResult response (both matching and nomatching mode) @ImmanuelSegol

### CODE REVIEW

File: lib/proto/google/api/http_pb.js [hunk 1]
```diff
@@ -153,15 +153,15 @@ proto.google.api.Http.serializeBinaryToWriter = function(message, writer) {
 
 /**
  * repeated HttpRule rules = 1;
- * @return {!Array.<!proto.google.api.HttpRule>}
+ * @return {!Array<!proto.google.api.HttpRule>}
```
Comment (erkarl) [inline lib/proto/google/api/http_pb.js]: I'm curious as to why we're getting diffs like this? @sangaman, is this related to the grpc-tools version changes?

Comment (sangaman) [reply lib/proto/google/api/http_pb.js]: Yes I think so, @ImmanuelSegol could you run `npm install` and then `npm run proto` again?

**Comment by ImmanuelSegol** [comment]
@kilrau @sangaman Also here the failed test are due to Travis, locally they work just fine.

**Comment by kilrau** [comment]
Restarted Travis build

**Comment by rsercano** [comment]
There's a test case in the GO tests which checks local id, that's why tests are failing; `order_matching_and_swap`

**Comment by sangaman** [comment]
@rsercano Do you need help modifying the go simulation test not to check the local id? It shoulnd't be too hard but if you're not sure where to start I can point you in the right direction. Thanks.

**Comment by rsercano** [comment]
Commented it out @sangaman, thanks a lot.

File: test/simulation/actions.go [hunk 2]
```diff
@@ -228,7 +228,7 @@ func (a *actions) placeOrderAndSwap(srcNode, destNode *xudtest.HarnessNode,
 
 	// Verify the swap events info.
 	a.assert.Equal(eMaker.swap.OrderId, eTaker.swap.OrderId)
-	a.assert.NotEqual(eMaker.swap.LocalId, eTaker.swap.LocalId)
+	//a.assert.NotEqual(eMaker.swap.LocalId, eTaker.swap.LocalId)
```
Comment (michael1011) [inline test/simulation/actions.go]: Why commenting it our instead of removing this line entirely?

Comment (rsercano) [reply test/simulation/actions.go]: I can remove it if you say so guys, can't imagine why would we require it @sangaman

**Comment by sangaman** [comment]
@kilrau I know the original idea behind this PR first came up many months ago, but I feel I was confused then as to its motive and am still confused now when reviewing this again - maybe it's just that we've made other changes that make this irrelevent.

If we give our orders local ids, we would want them on the `SwapSuccess` message that we get back from `SubscribeSwaps` and even from `PlaceOrder`. The *only* case where it clearly doesn't make sense to me to have a local id is in `nomatching` mode when calling `ExecuteSwap`. If that's the case, we can just ensure the local id is blank on the `ExecuteSwap` response. Does that make sense? If so I think it makes sense to close this PR and handle that in another one (I'm thinking it would be just a couple of lines).

**Comment by kilrau** [comment]
> Does that make sense? If so I think it makes sense to close this PR and handle that in another one (I'm thinking it would be just a couple of lines).

Probably. I'm closing this now and we'll look at it again once `nomatching` mode is a topic again. This has been dragged way too long.
